src/Microsoft.Health.Client/Extensions/IHttpClientBuilderRegistrationExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Client.Configuration;
+
+namespace Microsoft.Health.Client.Extensions;
+
+public static class IHttpClientBuilderRegistrationExtensions
+{
+    public static void AddAuthenticationHandler(this IHttpClientBuilder httpClientBuilder, IServiceCollection services, IConfigurationSection authenticationConfigurationSection, string credentialProviderName)
```

Comment:
We can make name optional, but I think I would want to make the default value a random value (probably a GUID). 

The reason being is that we're using that to resolve the credential provider and if we don't, we won't otherwise know which to select. The issue with the `httpClientBuilder.Name` is that it appears that if a name isn't specified, it's either going to be `Options.DefaultName` or the `TypeNameHelper.GetTypeDisplayName(typeof(TClient), fullName: false);` which can both have collisions.

Reply:
I am thinking for scenarios where there is only 1 set of credentials configured in this manner, it may make sense to leave it optional. And in those cases, I think the default name like `""` should be sufficient if there is only 1 configured credential provider.